spmd/tensor/device_mesh.py

```diff
@@ -249,24 +244,46 @@ def get_coordinate_on_dim(self, dim: int) -> Optional[int]:
         return self._coordinate_on_dim[dim] if self._coordinate_on_dim else None
 
     def scatter(
-        self, scatter_list: List[torch.Tensor], mesh_dim: int = 0
+        self,
+        tensor_to_scatter: torch.Tensor,
+        mesh_dim: int = 0,
+        tensor_dim: int = 0,
     ) -> torch.Tensor:
         """
-        scatter the list of tensors to a device mesh dimension. We by default
+        scatter a big tensor to a device mesh dimension. We by default
         use the first rank of the mesh dimension as the source of truth, i.e
         for a 2d mesh [[0, 1], [2, 3]], if we scatter on mesh_dim = 1, we will
-        scatter the tensor list on rank 0 to rank 0/1, and tensor list on rank 2
-        to rank 2/3.
+        scatter the tensor splitted on rank 0 to rank 0/1, and tensor splitted
+        on rank 2 to rank 2/3.
 
         Args:
-            scatter_list (List[torch.Tensor]): list of tensors to scatter.
+            tensor_to_scatter (torch.Tensor): the tensor to be scattered.
             mesh_dim (int, optional): indicate which mesh dimension we want
                 to scatter on, we by default choose the first rank on the
-                mesh dimension.
+                mesh dimension as source of truth.
+            tensor_dim (int, optional): indicate which tensor dimension we want
+                to split the `tensor_to_scatter` before scattering.
 
         Returns:
             A :class:`torch.Tensor` object
         """
+        my_coordinate = self.get_coordinate_on_dim(mesh_dim)
+        # TODO: what should happen if rank is not in the mesh?
+        assert (
+            my_coordinate is not None
+        ), "Rank if not part of mesh"  # TODO: figure out behavior here
```

Comment:
Should we just ignore non-participating ranks? I think it would be a consistent behavior, but we would have to do it throughout the code. (We can do this later).